src/js/main.js

```diff
@@ -0,0 +1,152 @@
+const filmButton = document.querySelector('#film-name-button');
+function handlerProperty(e){
+  console.log(e)
+  const films = ['Алиса в стране чудес', 'Мстители','Город мертвых', 'Призрачный гонщик']
+
+
+  // Посмотреть innerHTML. и обнулить контейнер
+  const filmsList = document.querySelector('.usefull-info-left__list');
+  filmsList.innerHTML = ''
+  // while (filmsList.firstChild) {                       Было / Стало
+  //   filmsList.removeChild(filmsList.firstChild);
+  // }
+  
+  
+  let input = document.getElementById('film-name-text').value ;
+  
+  
+  const filterFilmsList = films.filter( (film) => film.includes(input));
+  
+  const contentString =
+  filterFilmsList.map((item,index) => `
+      <li class="usefull-info-left__item">${item}</li>
+    `).join('') 
+  filmsList.insertAdjacentHTML('beforeend', contentString);
+}
+if(filmButton){
+  filmButton.addEventListener('click', handlerProperty )  
+}
+handlerProperty('click');
```

Comment:
тут конечно не все корректно, нужно за пределы функции вынести константы, при нажатии на кнопку, заново создается массив и ищется filmsList 

и нужно не забывать убирать комментарии

```suggestion
const filmButton = document.querySelector("#film-name-button");
const films = [
  "Алиса в стране чудес",
  "Мстители",
  "Город мертвых",
  "Призрачный гонщик",
];
const filmsList = document.querySelector(".usefull-info-left__list");
function handlerProperty() {
  let input = document.getElementById("film-name-text").value;
  filmsList.innerHTML = "";

  const filterFilmsList = films.filter((film) => film.includes(input));

  const contentString = filterFilmsList
    .map(
      (item, index) => `
      <li class="usefull-info-left__item">${item}</li>
    `
    )
    .join("");
  filmsList.innerHTML = contentString;
}
if (filmButton) {
  filmButton.addEventListener("click", handlerProperty);
}
handlerProperty();
```

Reply:
ну и insertAdjacentHTML тоже тут не нужен, мы же контейнер очищаем, он пустой, поэтому можем просто через innerHTML вставить новые узлы